src/test/groovy/fr/jcgay/maven/extension/revision/UniqueRevisionFilteringTest.groovy

```diff
@@ -61,14 +61,41 @@ class UniqueRevisionFilteringTest extends Specification {
         project.version == '1.0-SNAPSHOT'
     }
 
-    def 'replace ${revision} with current artifact version without parent pom'() {
+    def 'replace x.y.${revision} with current artifact version'() {
         given:
         pom << """<?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.github.jcgay.example.version</groupId>
+        <artifactId>parent-pom</artifactId>
+        <version>1.0.\${revision}</version>
+    </parent>
 
     <artifactId>submodule-1</artifactId>
+    <version>1.0.\${revision}</version>
+</project>
+        """
+        def artifact = new SubArtifact(new DefaultArtifact('fr.jcgay.test', 'jar-id', 'jar', '0-SNAPSHOT'), null, 'pom', pom)
```

Comment:
In real world, `version` in `DefaultArtifact` creation will be `1.0.0-SNAPSHOT`. This is the version resolved by Maven, not just the `${revision}` value.  
I don't remember why I have not added Integration Test(s) in this project (maybe I have failed to read the local repository) but there are lacking :)

Reply:
Yep. I agree. I ran into this problem this once I started using it this morning! 🤕 
For my purpose, I may need to source the revision value to resolve from a pom property.